MaterialDesignThemes.Wpf/Snackbar.cs

```diff
@@ -1,12 +1,40 @@
 ﻿using System;
+using System.Globalization;
 using System.Windows;
 using System.Windows.Controls;
+using System.Windows.Data;
 using System.Windows.Markup;
 using System.Windows.Media.Animation;
 using System.Windows.Threading;
 
 namespace MaterialDesignThemes.Wpf
 {
+    public enum SnackbarActionButtonPlacementMode
+    {
+        Auto,
+        Inline,
+        SeparateLine
+    }
+
+    public class SnackbarActionButtonPlacementModeConverter : IMultiValueConverter
+    {
+        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture)
+        {
+            var mode = (SnackbarActionButtonPlacementMode) values[0];
+            var snackbarHeight = (double) values[1];
+            return mode switch
+            {
+                SnackbarActionButtonPlacementMode.Auto when snackbarHeight > 53 => Dock.Bottom,
```

Comment:
Why the value 53? Is the the point where text wraps onto a second line?

Reply:
Yes, thats looks light some magic. It is the point, where the placement of the button is decided. For me I oriented me at the hint in the issue:

> We can detect this case by simply watching the height of the Content.

So I measured the height of a single lined snackbar message and it something below 53. And I simply took this as a threshold. But I do understand, that this looks somewhat arbitrary. Do you have an idea?

Reply:
Perhaps something like this. Create an attached property on SnackbarMessage (something like `SingleLineHeight`). Then set the attached property on the snackbar in its style and default it to 53. Finally pass it into the converter as a third parameter. 
That way we can attach a name to the value, and it allows someone to override it if needed.

Thoughts?

Reply:
I added `SnackbarMessage.InlineActionButtonMaxHeight` as you suggested.

Then I found the limiting `MaxHeight` on the `ContentPresenter`. I also added `SnackbarMessage.ContentMaxHeight` for this. But to make it consistent with the first property I moved its usage to the wrapping `DockPanel` and adapted the value. That way both values relates to the SnackbarMessage.